f/KAS-3431 Vrijgave: automanual

mdv-professional opened the pull request:
[Jira](https://kanselarij.atlassian.net/browse/KAS-3431)

Commit:
KAS-3431 data-model: convert release-(decisions/documents) to internal-(decision/document)-publication-activity
- resource definition
- migration
Release decisions: Automanual

Commit:
KAS-3431 migration: convert in kanselarij graph + delete old from all graphs
Release decisions: Automanual

Commit:
KAS-3431 repurpose XPublicationActivity.xStartTime
Release decisions: Automanual

Commit:
KAS-3431 fix: repurpose XPublicationActivity.xStartTime: consistent predicate use
Release decisions: Automanual

Commit:
KAS-3431 remove publicationDate and publicationDocDate / /config/resource/newsletter-domain
Release decisions: Automanual

ValenberghsSven committed:
Merge branch 'development' of github.com:kanselarij-vlaanderen/app-kaleidos into f/KAS-3431-release--automanual

ValenberghsSven committed:
Merge branch 'f/KAS-3431-release--automanual' of github.com:kanselarij-vlaanderen/app-kaleidos into f/KAS-3431-release--automanual

ValenberghsSven commented on config/migrations/20220621162039-insert-internal-x-publication-activities-kanselarij.sparql:
```diff
@@ -0,0 +1,41 @@
+        ?decisionPub prov:startedAtTime ?publicationTime .
+    }
+} WHERE {
+    GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
+        ?meeting a besluit:Vergaderactiviteit .
+        ?meeting ext:releasedDecisions ?publicationTime .
```
op productie vallen 4 vergaderingen buiten deze query
```
PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
PREFIX prov: <http://www.w3.org/ns/prov#>
PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
PREFIX besluit: <http://data.vlaanderen.be/ns/besluit#>
PREFIX besluitvorming: <http://data.vlaanderen.be/ns/besluitvorming#>


select * WHERE {
  GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
    ?meeting a besluit:Vergaderactiviteit .
    ?meeting <http://data.vlaanderen.be/ns/besluit#geplandeStart> ?o .
    FILTER NOT EXISTS { ?meeting ext:releasedDecisions ?publicationTime .}
  }
}
```
2 ervan staan op themis en zijn EP agendas.
De andere 2 staan niet op themis.

Op zich geen probleem denk ik, als de  beslissingen niet zijn vrijgegeven via onze app dan hoeft er ook geen activity te bestaan.
Maar dan moeten we in de frontend ook rekening houden met het feit dat dit dus optioneel is.

Ik bedenk me dan ook:
Met de vooraf aangemaakte modellen, is er dan nog de mogelijkheid om nooit beslissingen/documenten vrij te geven?

ValenberghsSven commented on config/migrations/20220621162039-insert-internal-x-publication-activities-kanselarij.sparql:
```diff
@@ -0,0 +1,41 @@
+        ?documentPub ext:internalDocumentPublicationActivityUsed ?meeting .
+        ?documentPub prov:startedAtTime ?publicationTime .
+    }
+} WHERE {
+    GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
+        ?meeting a besluit:Vergaderactiviteit .
+        ?meeting ext:releasedDocuments ?publicationTime .
```
dit zijn er een pak meer, 12 agendas waar geen documenten voor zijn vrijgegeven (intern of extern dus)

ValenberghsSven replied:
Moeten we voor die vergadering toch ook modellen maken, maar dan zonder `prov:startedAtTime` om te reflecteren dat ze nooit vrijgegeven zijn geweest /gaan worden?

ValenberghsSven commented on config/resources/domain.lisp:
```diff
@@ -12,7 +12,7 @@
    of responses in the result object's meta.")
 (defparameter *max-group-sorted-properties* nil)
-(defparameter sparql:*query-log-types* nil)
+; (defparameter sparql:*query-log-types* nil)
```
dit niet mergen

ValenberghsSven commented on config/resources/newsletter-domain.lisp:
```diff
@@ -54,10 +52,38 @@
 
+(define-resource internal-decision-publication-activity () ; FIXME extend prov:Activity
+  :class (s-prefix "ext:InternalDecisionPublicationActivity")
+  :properties `((:start-time       :datetime     ,(s-prefix "prov:startedAtTime"))
```
moet deze fixme `extend prov:Activity`?
Staat niet in ticket vermeld. Willen we dat nu doen?
Gaat mogelijk wel botsen met logica in `themis-export-service` want die insert/select `prov:Activity`
We kunnen al een `dct:type` voorzien voor de nieuwe soorten.
Polymorphisme van de relatie `prov:used` gaat ons mogelijk te ver leiden

ValenberghsSven commented on config/resources/besluit-domain.lisp:
```diff
@@ -207,7 +206,16 @@
              (meeting                   :via      ,(s-prefix "dct:isPartOf")
-                                        :as "main-meeting"))
+                                        :as "main-meeting")
+             (internal-decision-publication-activity :via  ,(s-prefix "ext:internalDecisionPublicationActivityUsed")
+                                        ; optional:  present in new meetings since KAS-3431 / optional in older meetings
+                                        :inverse t
+                                        :as "internal-decision-publication-activity")
+             (internal-document-publication-activity :via  ,(s-prefix "ext:internalDocumentPublicationActivityUsed")
+                                        ; optional: present in new meetings since KAS-3431 / optional in older meetings
+                                        :inverse t
```
Zoals eerder gezegd, zouden we deze wel optioneel maken? We kunnen er ook maken zonder start datums.
Op die manier kunnen ze altijd later nog ingesteld worden en is consistenter om te gebruiken in frontend (ipv `null or undefined`). Ook als we bvb vrijdag avond zouden releasen dan heeft de agenda nog geen "releasedDocuments" flag (want dat komt pas maandag), dus die meeting mist dan ineens dat model toch?.

ValenberghsSven committed:
KAS-3431 improve readability of lisp file by alligning and remove whitespace

ValenberghsSven commented on config/resources/newsletter-domain.lisp:
```diff
@@ -54,12 +52,35 @@
+  :properties `((:start-time       :datetime     ,(s-prefix "prov:startedAtTime")))
+  :has-one `((meeting              :via          ,(s-prefix "ext:internalDecisionPublicationActivityUsed") ; FIXME prov:used / workaround for meeting having multiple relationships of a subtype of prov:Activity
+                                   :as "meeting"))
+  :resource-base (s-url "http://themis.vlaanderen.be/id/internebeslissingspublicatieactiviteit/")
```
Willen we deze themis uri op deze manier of met streepjes tussen (lijkt meer consistent met andere, bvb `"http://themis.vlaanderen.be/id/themis-publicatie-activiteit/"`

ValenberghsSven commented on config/resources/newsletter-domain.lisp:
```diff
@@ -3,8 +3,6 @@
   :properties `((:text                  :string   ,(s-prefix "besluitvorming:inhoud"))
                 (:richtext              :string   ,(s-prefix "ext:htmlInhoud"))
                 (:subtitle              :string   ,(s-prefix "dbpedia:subtitle"))
-                (:publication-date      :datetime ,(s-prefix "dct:issued"))
-                (:publication-doc-date  :datetime ,(s-prefix "ext:issuedDocDate"))
```
Heb ik dit gedaan? leek me te komen van een merge.
Ik denk dat dit moet blijven staan

ValenberghsSven replied:
mocht uiteindelijk toch weg

ValenberghsSven committed:
KAS-3431 revive some attributes of newsletter

ValenberghsSven commented:
de migratie heeft rare gevolgen.
Hij koppelt alle activiteiten aan alle meetings...
![image](https://user-images.githubusercontent.com/22245223/179046215-0cc4ea12-ea54-4609-bf58-44e5d37441e3.png)

ValenberghsSven commented:
Lijkt beter als ik met subquery werk., ik weet niet waarom die zoveel zaken matched aan elkaar met de gewone query.
DIe queries zijn ok in sparql endpoint, maar in migration-service gaat het mis.

ValenberghsSven committed:
KAS-3431 combined commit

add release status, change migration to due mismapping error, add migration for active meetings, remove unconfirmed date, rename startTime back to startDate

ValenberghsSven commented on config/migrations/20220714120100-insert-internal-x-publication-activities-kanselarij.sparql:
```diff
@@ -0,0 +1,107 @@
+
+;
+
+INSERT {
+  GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
+    ?themisPublicationActivity a ext:ThemisPublicationActivity ;
+      mu:uuid ?newThemisPublicationuuid ;
+      prov:used ?meeting ;
+      ext:scope "newsitems" ;
+      ext:scope "documents" ;
+      generiek:geplandeStart ?plannedDocPublicationDate ;
+      adms:status <http://themis.vlaanderen.be/id/concept/vrijgave-status/1bd5b05a-07af-46d3-827f-297f993b8b54> .
+  }
+}
+WHERE {
+  GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
+    ?meeting a besluit:Vergaderactiviteit .
+    {
+      SELECT DISTINCT ?meeting (STRUUID() AS ?newThemisPublicationuuid) ?plannedDocPublicationDate WHERE {
+        GRAPH <http://mu.semte.ch/graphs/organizations/kanselarij> {
+          ?meeting a besluit:Vergaderactiviteit ;
+            ext:algemeneNieuwsbrief ?newsletter ;
+            ext:finaleZittingVersie "false"^^<http://mu.semte.ch/vocabularies/typed-literals/boolean> .
+          ?newsletter ext:issuedDocDate ?plannedDocPublicationDate .
+        }
+      }
+    }
+    BIND(IRI(CONCAT("http://themis.vlaanderen.be/id/themis-publicatie-activiteit/", ?newThemisPublicationuuid)) AS ?themisPublicationActivity)
+  }
+}
+
```
Ik had deze query gemaakt om voor alle openstaande meetings ook een themis-publication aan te maken.
DIt is vooral voor cypress test data.
In praktijk (release in het weekend) gaat deze query niet de agenda opvangen die al is afgesloten op vrijdag, maar de documenten nog niet zijn vrijgegeven.
Daar moet nog een aparte query voor gemaakt worden, of deze aangepast.

ValenberghsSven committed:
KAS-3431 add and remove comments

erikap committed:
Merge branch 'development' into f/KAS-3431-release--automanual

erikap committed:
Merge branch 'f/KAS-3431-release--automanual' of github.com:kanselarij-vlaanderen/app-kaleidos into f/KAS-3431-release--automanual

erikap committed:
Use generic skos:Concept to model release status on publication activities

erikap committed:
Revert enabling of SPARQL query logging in mu-cl-resources

erikap committed:
Convert publication activity models from Lisp to JSON

erikap committed:
Remove custom subclass from release statusses migration

erikap committed:
Split migrations to insert publication activities for existing meetings

erikap committed:
Fix typo in migration

erikap committed:
Fix errors in meeting publication activity models

erikap committed:
Rename domain file for meeting-activities

erikap committed:
Fix typo in meeting-activities domain model

erikap committed:
Use feature branches for automanual document release

erikap committed:
Delete duplicate migration file

erikap committed:
Only insert pub activity if it doesn't exist yet on meeting

erikap committed:
Merge branch 'development' into f/KAS-3431-release--automanual

erikap committed:
Use tagged releases of upgraded services

erikap committed:
Update migration to insert document/themis document publication activities
